### PR TITLE
Split out control API into its own chapter

### PR DESCRIPTION
**Pull request by badboy**
https://bugzilla.mozilla.org/show_bug.cgi?id=1604153

---

This is mostly reorganizing the existing content.
Given that we don't yet have documentation for the internal or external Rust API, it looks pretty much the same across language wrappers. :)

### CODE REVIEW

**Comment by badboy** [comment]
Oh, this also contains one slightly unrelated commit that moves one section down so it matches the sections for the other languages.

**Comment by badboy** [comment]
Here's what it looks like rendered:

image

File: docs/user/control-api.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Controlling Glean
```
Comment (Dexterp37) [inline docs/user/control-api.md]: Note: this is not the "control API", this is the "general API" (in contrast with the "specific metric API")

File: docs/user/control-api.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+[custom pings]: pings/custom.md
+[experiments-api]: experiments-api.md
+
+The following steps are required for applications using the Glean SDK, but not libraries.
+
+> **Important:** The Glean SDK should only be initialized from the main application, not individual libraries.
+
+If you are adding Glean support to a library, you can safely skip this section.
+
+## General control API
```
Comment (Dexterp37) [inline docs/user/control-api.md]:
```suggestion
## The API
```

File: docs/user/control-api.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+The Glean SDK provides a general control API, that supports the following operations. See below for language-specific details.
```
Comment (Dexterp37) [inline docs/user/control-api.md]:
```suggestion
The Glean SDK provides a general API that supports the following operations. See below for language-specific details.
```

File: docs/user/control-api.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
+# Controlling Glean
+
+Glean has a minimal control API available on its top-level `Glean` object.
+This API allows to enable and disable upload, register [custom pings][] and set [experiment data][experiments-api].
```
Comment (Dexterp37) [inline docs/user/control-api.md]: This is missing the link for 'custom pings'.. but it seems to work anyway!? Interesting.

File: docs/user/control-api.md (hunk 1 above)
Comment (Dexterp37) [inline docs/user/control-api.md]:
```suggestion
# The General API
```

File: docs/SUMMARY.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 - [Glean](README.md)
 - [Using the Glean SDK](user/index.md)
     - [Adding Glean to your project](user/adding-glean-to-your-project.md)
+    - [Controlling Glean](user/control-api.md)
```
Comment (Dexterp37) [inline docs/SUMMARY.md]:
```suggestion
    - [The General API](user/control-api.md)
```

File: docs/user/control-api.md (hunk 4 above)
Comment (travis79) [reply docs/user/control-api.md]: Is it doing something smart like matching the name with the reference below?

Comment (Dexterp37) [reply docs/user/control-api.md]: Good question, maybe? I would have imagined this to fail, given how `experiment data` is linked!

File: docs/user/control-api.md (hunk 1 above)
Comment (badboy) [reply docs/user/control-api.md]: And here I thought I could slip that in.

File: docs/user/control-api.md (hunk 4 above)
Comment (badboy) [reply docs/user/control-api.md]: It is! Short hand markdown form (I had that for the "experiments data" before and changed it, let me unify this here.

```suggestion
This API allows to enable and disable upload, register [custom pings][custom-pings] and set [experiment data][experiments-api].
```

File: docs/user/control-api.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+[custom pings]: pings/custom.md
```
Comment (badboy) [inline docs/user/control-api.md]:
```suggestion
[custom-pings]: pings/custom.md
```

File: docs/SUMMARY.md [hunk 7]
```diff
@@ -3,6 +3,7 @@
 - [Glean](README.md)
 - [Using the Glean SDK](user/index.md)
     - [Adding Glean to your project](user/adding-glean-to-your-project.md)
+    - [The General API](user/control-api.md)
```
Comment (Dexterp37) [inline docs/SUMMARY.md]: To be consistent, would you kindly rename the file to `general-api.md`?

File: docs/user/control-api.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
+# The General API
+
+Glean has a minimal control API available on its top-level `Glean` object.
```
Comment (Dexterp37) [inline docs/user/control-api.md]:
```suggestion
Glean has a minimal API available on its top-level `Glean` object.
```

**Comment by badboy** [comment]
Using admin power to land this, integration test is hopefully fixed soon.

**Pull request merged**